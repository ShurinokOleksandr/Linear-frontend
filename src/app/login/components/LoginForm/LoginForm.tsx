'use client';
import { CustomLink, Typography, LogoIcon, Button, Input, Title, Span, Box } from '@/Components/ui';
import styled, { useTheme } from 'styled-components';
import { useRouter } from 'next/navigation';
import { observer } from 'mobx-react-lite';
import React from 'react';

import loginFormStore from '../../model/store';

export const LoginForm =  observer(() => {
	 
 	const theme = useTheme();
	 
	const router = useRouter();
	 
	const loginFn = async () =>  {
		const res = await loginFormStore.submitLogin();
 		if(res){
			router.push('/');
		}else{
			return null;
		}
	};
  	return (
		<Wrapper>
			<Logo>
				<LogoIcon />
			</Logo>
			<Title
				data-testid='titleForm'
				textAlign='center'
				fontSize='20px'
			>
				Log in to Linear
			</Title>
			<Box >
				<Input
					onChange={(e:React.ChangeEvent<HTMLInputElement>) => loginFormStore.setUsername(e.target.value)}
					pointerEvents={loginFormStore.loading ? 'none' : 'auto'}
					placeholder='Enter your Name...'
					fontSize={theme.fontSizes.small}
					marginBottom='7px'
					display='block'
					padding='10px'
					height='46px'
				/>
				{
					loginFormStore.usernameValidationError && <Typography
						data-testid='usernameError'
						color={theme.critical}
						fontSize='12px'
					>
						{loginFormStore.usernameValidationError}
					</Typography>
				}
			</Box>
			<Box>
				<Input
					onChange={(e:React.ChangeEvent<HTMLInputElement>) => loginFormStore.setPassword(e.target.value)}
					pointerEvents={loginFormStore.loading ? 'none' : 'auto'}
 					placeholder='Enter your Password...'
					fontSize={theme.fontSizes.small}
					marginBottom='7px'
					type='password'
					display='block'
					padding='10px'
					height='46px'
				/>
				{
					loginFormStore.usernameValidationError && <Typography
						data-testid='passwordError'
						color={theme.critical}
						fontSize='12px'
					>
						{loginFormStore.passwordValidationError}
					</Typography>
				}
			</Box>
			<Button
 				cursor={loginFormStore.loading ? 'auto' : 'pointer' }
				disabled={loginFormStore.loading}
				fontSize={theme.fontSizes.small}
				borderRadius='4px'
				textAlign='center'
				onClick={loginFn}
				display='block'
				height='48px'
				width='100%'
			>
				Login
			</Button>
			<Box
				fontSize='12px'
			>
				<Span marginRight='5px'>
					Do you have not an account?
				</Span>
				<CustomLink color={theme.purple300} href='/signup'>
					Sign Up
				</CustomLink>
			</Box>
		</Wrapper>
	);
});
const Wrapper  = styled.main`
  color: ${({ theme }) => theme.gray200};
  display: flex;
  flex-direction: column;
  gap: 20px;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  height: 100%;
  width: 336px;
`;
const Logo = styled.div`
  width: 84px;
  height: 84px;
`;

