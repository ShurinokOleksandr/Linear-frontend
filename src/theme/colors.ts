export const colors = {
	scrim: 'rgba(0, 0, 0, 0.60)',
	surface3_light: '#22222212',
	surface4_light: '#FFFFFF64',
	surface5_light: '#00000004',
	neutral1_light: '#222222',
	neutral2_light: '#7D7D7D',
	neutral3_light: '#CECECE',
	surface1_light: '#FFFFFF',
	surface2_light: '#F9F9F9',
	accent1_light: '#FC72FF',
	accent2_light: '#FFEFFF',
	purple100: '#6C79FF',
	purple150: '#7477F0',
	purple200: '#6E79D6',
	purple250: '#5E6AD2',
	purple300: '#666BE2',
	purple350: '#5C67C7',
	purple400: '#575BC7',
	purple600: '#37466C',
	purple700: '#2A2B51',
	purple800: '#222342',
	critical: '#FF5F52',
	gray100: '#EEEFFC',
	gray150: '#E0E1EC',
	gray200: '#D2D3E0',
	gray250: '#B0B5C0',
	gray300: '#858699',
	gray350: '#6B6F76',
	gray400: '#8D8BA3',
	gray450: '#858699',
	gray500: '#4C4F6B',
	gray550: '#4D4F69',
	gray600: '#444556',
	gray650: '#3C3D53',
	gray700: '#393A4B',
	gray750: '#313248',
	gray800: '#292B41',
	gray850: '#26293E',
	gray900: '#2C2D42',
	gray950: '#26273B',
	success: '#40B66B',
	gray960:'#292A35',
	gray970:'#292A35',
	gray50: '#F5F6FC',
	white: '#FFFFFF',
	black: '#000000',
	teal:'#00B2BF',
};
const commonTheme = {
	purple100: colors.purple100,
	purple150: colors.purple150,
	purple200: colors.purple200,
	purple250: colors.purple250,
	purple300: colors.purple300,
	purple350: colors.purple350,
	purple400: colors.purple400,
	purple600: colors.purple600,
	purple700: colors.purple700,
	purple800: colors.purple800,
	gray100: colors.gray100,
	gray150: colors.gray150,
	gray200: colors.gray200,
	gray250: colors.gray250,
	gray300: colors.gray300,
	gray350: colors.gray350,
	gray400: colors.gray400,
	gray450: colors.gray450,
	gray500: colors.gray500,
	gray550: colors.gray550,
	gray600: colors.gray600,
	gray650: colors.gray650,
	gray700: colors.gray700,
	gray750: colors.gray750,
	gray800: colors.gray800,
	gray850: colors.gray850,
	gray900: colors.gray900,
	gray950: colors.gray950,
	gray960: colors.gray960,
	gray970: colors.gray970,
	gray50: colors.gray50,
	white: colors.white,
	black: colors.black,
	teal: colors.teal,
};
export const lightTheme = {
	...commonTheme,
	neutral1: colors.neutral1_light,

	neutral2: colors.neutral2_light,
	neutral3: colors.neutral3_light,
	surface1: colors.surface1_light,

	surface2: colors.surface2_light,
	surface3: colors.surface3_light,
	surface4: colors.surface4_light,
	surface5: colors.surface5_light,
	accent1: colors.accent1_light,

	accent2: colors.accent2_light,
	critical: colors.critical,

	background: colors.white,
	success: colors.success,
	scrim: colors.scrim,
};
export const darkTheme = {
	...commonTheme,
	neutral2: colors.neutral2_light,
	neutral3: colors.neutral3_light,
	surface1: colors.surface1_light,

	accent1: colors.accent1_light,
	accent2: colors.accent2_light,
	background: colors.gray900,
	critical: colors.critical,
	surface2: colors.gray550,

	surface3: colors.gray650,
	surface4: colors.gray600,

	surface5: colors.gray500,
};