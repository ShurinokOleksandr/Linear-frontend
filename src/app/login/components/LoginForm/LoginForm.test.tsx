import { LoginForm } from '@/app/login/components/LoginForm/LoginForm';
import { cleanup, render,screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { darkTheme } from '@/theme/colors';
import { afterEach } from '@jest/globals';
import '@testing-library/jest-dom';

afterEach(cleanup);

jest.mock('next/navigation', () => ({
	useRouter() {
		return {
			prefetch: () => null
		};
	}
}));

describe('Testing login form component', () => {
	it('should render', () => {
		render(<ThemeProvider theme={darkTheme} ><LoginForm /></ThemeProvider> );
		const inputUsername = screen.getByPlaceholderText('Enter your Name...');
		const inputPassword = screen.getByPlaceholderText('Enter your Password...');
		const btn = screen.getByRole('button');
 		expect(inputPassword).toBeInTheDocument();
		expect(inputUsername).toBeInTheDocument();
		expect(btn).toBeInTheDocument();
		screen.debug();
		// expect(getByTestId('titleForm').textContent).toBe('Log in to Linear');
	});
});
