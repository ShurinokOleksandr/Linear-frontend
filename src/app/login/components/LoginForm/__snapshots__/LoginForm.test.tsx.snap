// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login form component should render 1`] = `
<DocumentFragment>
  <main
    class="LoginForm__Wrapper-sc-1yxwupj-0 fuyJHG"
  >
    <div
      class="LoginForm__Logo-sc-1yxwupj-1 cvEvDK"
    >
      <svg
        class="sc-jlqnSw eEomXO"
        fill="none"
        height="84"
        viewBox="0 0 1024 1024"
        width="84"
      >
        <rect
          fill="url(#linear-icon-a)"
          height="100%"
          rx="9999"
          width="100%"
        />
        <g
          filter="url(#linear-icon-b)"
        >
          <path
            d="M262.274 570.997c-1.139-4.857 4.646-7.915 8.174-4.388l186.943 186.943c3.527 3.528.469 9.313-4.388 8.174-94.339-22.131-168.598-96.39-190.729-190.729Z"
            fill="url(#linear-icon-c)"
          />
          <path
            d="M256.01 496.072c-.091 1.451.455 2.867 1.482 3.894l266.542 266.542c1.027 1.027 2.443 1.573 3.894 1.482 12.13-.755 24.032-2.355 35.647-4.74 3.914-.804 5.274-5.613 2.449-8.439L269.189 457.976c-2.826-2.825-7.635-1.465-8.439 2.449-2.385 11.615-3.985 23.517-4.74 35.647Z"
            fill="url(#linear-icon-d)"
          />
          <path
            d="M277.56 408.092c-.852 1.914-.418 4.15 1.063 5.631l331.654 331.654c1.481 1.481 3.717 1.915 5.631 1.063 9.145-4.073 18.008-8.666 26.55-13.742 2.827-1.68 3.263-5.564.938-7.889L299.191 380.604c-2.325-2.325-6.209-1.889-7.889.938-5.076 8.542-9.669 17.405-13.742 26.55Z"
            fill="url(#linear-icon-e)"
          />
          <path
            d="M320.813 348.539c-1.895-1.895-2.013-4.935-.227-6.933C367.511 289.072 435.77 256 511.754 256 653.275 256 768 370.725 768 512.246c0 75.984-33.072 144.243-85.606 191.168-1.998 1.786-5.038 1.668-6.933-.227L320.813 348.539Z"
            fill="url(#linear-icon-f)"
          />
        </g>
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="linear-icon-a"
            x1="512"
            x2="512"
            y1="0"
            y2="1024"
          >
            <stop
              stop-color="#5C6BF1"
            />
            <stop
              offset="1"
              stop-color="#283188"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="linear-icon-c"
            x1="274.286"
            x2="541.257"
            y1="303.543"
            y2="694.857"
          >
            <stop
              stop-color="#fff"
            />
            <stop
              offset="1"
              stop-color="#fff"
              stop-opacity=".65"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="linear-icon-d"
            x1="274.286"
            x2="541.257"
            y1="303.543"
            y2="694.857"
          >
            <stop
              stop-color="#fff"
            />
            <stop
              offset="1"
              stop-color="#fff"
              stop-opacity=".65"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="linear-icon-e"
            x1="274.286"
            x2="541.257"
            y1="303.543"
            y2="694.857"
          >
            <stop
              stop-color="#fff"
            />
            <stop
              offset="1"
              stop-color="#fff"
              stop-opacity=".65"
            />
          </lineargradient>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="linear-icon-f"
            x1="274.286"
            x2="541.257"
            y1="303.543"
            y2="694.857"
          >
            <stop
              stop-color="#fff"
            />
            <stop
              offset="1"
              stop-color="#fff"
              stop-opacity=".65"
            />
          </lineargradient>
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="797.149"
            id="linear-icon-b"
            width="797.149"
            x="113.426"
            y="135.703"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
            <fecolormatrix
              in="SourceAlpha"
              result="hardAlpha"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            />
            <feoffset
              dy="22.2772"
            />
            <fegaussianblur
              stdDeviation="71.2871"
            />
            <fecolormatrix
              values="0 0 0 0 0.118924 0 0 0 0 0.158031 0 0 0 0 0.570833 0 0 0 0.7 0"
            />
            <feblend
              in2="BackgroundImageFix"
              result="effect1_dropShadow_9417_5600"
            />
            <fecolormatrix
              in="SourceAlpha"
              result="hardAlpha"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
            />
            <feoffset
              dy="7.79703"
            />
            <fegaussianblur
              stdDeviation="30"
            />
            <fecolormatrix
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.17 0"
            />
            <feblend
              in2="effect1_dropShadow_9417_5600"
              result="effect2_dropShadow_9417_5600"
            />
            <feblend
              in="SourceGraphic"
              in2="effect2_dropShadow_9417_5600"
              result="shape"
            />
          </filter>
        </defs>
      </svg>
    </div>
    <h1
      class="CommonProps-sc-6h1a0q-0 Title-sc-1m9516y-0 lkaoeC HtyJI"
      font-size="20px"
      textalign="center"
    >
      Log in to Linear
    </h1>
    <div
      class="CommonProps-sc-6h1a0q-0 Box-sc-t5z15p-0 jjoQwq gMnMUV"
    >
      <input
        class="CommonProps-sc-6h1a0q-0 Input-sc-mrirja-0 jwlJIY bGLeAu"
        display="block"
        font-size="0.8125rem"
        height="46px"
        marginbottom="7px"
        padding="10px"
        placeholder="Enter your Name..."
        pointer-events="auto"
      />
    </div>
    <div
      class="CommonProps-sc-6h1a0q-0 Box-sc-t5z15p-0 jjoQwq gMnMUV"
    >
      <input
        class="CommonProps-sc-6h1a0q-0 Input-sc-mrirja-0 jwlJIY bGLeAu"
        display="block"
        font-size="0.8125rem"
        height="46px"
        marginbottom="7px"
        padding="10px"
        placeholder="Enter your Password..."
        pointer-events="auto"
        type="password"
      />
      <p
        class="CommonProps-sc-6h1a0q-0 Typography-sc-cv6znx-0 cmHaCf"
        color="#eb5757"
        font-size="12px"
      />
    </div>
    <button
      borderradius="4px"
      class="CommonProps-sc-6h1a0q-0 Button-sc-17pp0v7-0 fMwZFS fuCMtK"
      cursor="pointer"
      display="block"
      font-size="0.8125rem"
      height="48px"
      textalign="center"
      width="100%"
    >
      Login
    </button>
    <div
      class="CommonProps-sc-6h1a0q-0 Box-sc-t5z15p-0 bxrXrI gMnMUV"
      font-size="12px"
    >
      <span
        class="CommonProps-sc-6h1a0q-0 Span-sc-1sigtr0-0 jHJHBH fvAYZa"
        marginright="5px"
      >
        Do you have not an account?
      </span>
      <a
        class="CommonProps-sc-6h1a0q-0 CustomLink-sc-yyrbh9-0 dkrXQN kepjLa"
        color="#666BE2"
        href="/signup"
      >
        Sign Up
      </a>
    </div>
  </main>
</DocumentFragment>
`;
